sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py

```diff
@@ -744,10 +755,22 @@ async def get_throughput(self, **kwargs: Any) -> ThroughputProperties:
         if response_hook:
             response_hook(self.client_connection.last_response_headers, throughput_properties)
 
+        if 'offerAutopilotSettings' in throughput_properties[0]['content'] and 'autoUpgradePolicy' in \
+                throughput_properties[0]['content']['offerAutopilotSettings']:
+            return ThroughputProperties(properties=throughput_properties[0], auto_scale_max_throughput= \
+                throughput_properties[0]['content']['offerAutopilotSettings']['maxThroughput'],
+                                        auto_scale_increment_percent=throughput_properties[0]['content']
+                                        ['offerAutopilotSettings']['autoUpgradePolicy']['throughputPolicy'][
+                                            'incrementPercent'])
+
+        elif 'offerAutopilotSettings' in throughput_properties[0]['content']:
+            return ThroughputProperties(properties=throughput_properties[0],
+                                        auto_scale_max_throughput=throughput_properties[0]['content']
+                                        ['offerAutopilotSettings']['maxThroughput'])
+
```

Comment:
This also being duplicated is particularly problematic because it is very hard to follow as written.

Reply:
The reason why this code is being repeated is because of the same thing I said before we want to have the async match the sync version of the code.